On Windows, default rust needs to be gnu, not msvc

Apparently, under some circumstances, the default Rust on Windows needs to be gnu, not msvc.

See:
https://github.com/extendr/rextendr/issues/15#issuecomment-757944529
and
https://github.com/extendr/rextendr/pull/7#issuecomment-758272287

Is there any way to remove this requirement?

cc @Ilia-Kosenkov @CGMossa @yutannihilation

I did share it, see the link that Claus posted, or here:

```
error[E0463]: can't find crate for `std`                       
  |
  = note: the `x86_64-pc-windows-gnu` target may not be installed
```

While I'm at it: Using `cargo +stable-gnu build` does not work, as I think you can only write few things in `cargo +<here>`.

Ok.

I added `rustup target add x86_64-pc-windows-gnu` and somehow that added it to `stable-msvc`. This made
a new error appear: 

```
error: linker `x86_64-w64-mingw32-gcc` not found
  |
  = note: The system cannot find the file specified. (os error 2)
```

Then I added gcc that comes from rtools onto the path;  Via

```powershell
PS C:\Users\angus> $env:Path += ";C:\rtools40\usr\bin\;C:\rtools40\mingw64\bin\";
```

 And then I get another error:

```
[omitted]
  = note: C:/rtools40/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lgcc_eh
          collect2.exe: error: ld returned 1 exit status
```

And now I think this is just not possible to get past.

```r
> Sys.which("ld")
                                  ld 
"C:\\rtools40\\mingw64\\bin\\ld.exe"
```

That's it from me; I don't know what else to do about this.

`cannot find -lgcc_eh` is normally the problem when you're **not** making msvc the default toolchain. Weird. Maybe some problem with paths that get set when R is running. It should not be using the rtools linker but instead the msvc one, I believe.

--- a/R/source.R
+++ b/R/source.R
@@ -12,7 +12,8 @@
 #' @param profile Rust profile. Can be either `"dev"` or `"release"`. The default,
 #'   `"dev"`, compiles faster but produces slower code.
 #' @param toolchain Rust toolchain. The default, `NULL`, compiles with the
-#'  system default toolchain. Accepts valid Rust toolchain qualifiers.
+#'  system default toolchain. Accepts valid Rust toolchain qualifiers,
+#'  such as `"nightly"`, or (on Windows) `"stable-msvc"`.
 #' @param extendr_version Version of the extendr-api crate, provided as a Rust
 #'   version string. `"*"` will use the latest available version on crates.io.
 #' @param extendr_macros_version Version of the extendr-macros crate, if different

--- a/R/source.R
+++ b/R/source.R
@@ -11,6 +11,8 @@
 #'   be added to the `Cargo.toml` file.
 #' @param profile Rust profile. Can be either `"dev"` or `"release"`. The default,
 #'   `"dev"`, compiles faster but produces slower code.
+#' @param toolchain Rust toolchain. The default, `NULL`, compiles with the
+#'  system default toolchain. Accepts valid Rust toolchain qualifiers.
 #' @param extendr_version Version of the extendr-api crate, provided as a Rust
 #'   version string. `"*"` will use the latest available version on crates.io.
 #' @param extendr_macros_version Version of the extendr-macros crate, if different

--- a/R/source.R
+++ b/R/source.R
@@ -77,7 +77,9 @@ rust_source <- function(file, code = NULL, dependencies = NULL,
                           'extendr-api = { git = "https://github.com/extendr/extendr" }',
                           'extendr-macros = { git = "https://github.com/extendr/extendr" }'
                         ),
-                        profile = c("dev", "release"), extendr_version = "*",
+                        profile = c("dev", "release"),
+                        toolchain = NULL,
+                        extendr_version = "*",
                         extendr_macros_version = extendr_version,
                         env = parent.frame(),
                         use_extendr_api = TRUE,
@@ -124,6 +126,7 @@ rust_source <- function(file, code = NULL, dependencies = NULL,
   status <- system2(
     command = "cargo",
     args = c(
+      sprintf("+%s", toolchain),
       "build",
       "--lib",
       if (!is.null(specific_target)) sprintf("--target %s", specific_target) else NULL,

--- a/man/rust_source.Rd
+++ b/man/rust_source.Rd
@@ -39,7 +39,8 @@ be added to the \code{Cargo.toml} file.}
 \code{"dev"}, compiles faster but produces slower code.}
 
 \item{toolchain}{Rust toolchain. The default, \code{NULL}, compiles with the
-system default toolchain. Accepts valid Rust toolchain qualifiers.}
+system default toolchain. Accepts valid Rust toolchain qualifiers,
+such as \code{"nightly"}, or (on Windows) \code{"stable-msvc"}.}
 
 \item{extendr_version}{Version of the extendr-api crate, provided as a Rust
 version string. \code{"*"} will use the latest available version on crates.io.}

--- a/man/rust_source.Rd
+++ b/man/rust_source.Rd
@@ -13,6 +13,7 @@ rust_source(
     patch.crates_io = c("extendr-api = { git = \\"https://github.com/extendr/extendr\\" }",
     "extendr-macros = { git = \\"https://github.com/extendr/extendr\\" }"),
   profile = c("dev", "release"),
+  toolchain = NULL,
   extendr_version = "*",
   extendr_macros_version = extendr_version,
   env = parent.frame(),
@@ -36,6 +37,9 @@ be added to the \code{Cargo.toml} file.}
 
 \item{profile}{Rust profile. Can be either \code{"dev"} or \code{"release"}. The default,
 \code{"dev"}, compiles faster but produces slower code.}
+
+\item{toolchain}{Rust toolchain. The default, \code{NULL}, compiles with the
+system default toolchain. Accepts valid Rust toolchain qualifiers.}
 
 \item{extendr_version}{Version of the extendr-api crate, provided as a Rust
 version string. \code{"*"} will use the latest available version on crates.io.}

